elide-annotations/src/main/java/com/yahoo/elide/security/User.java

```diff
@@ -1,19 +1,28 @@
 /*
- * Copyright 2016, Yahoo Inc.
+ * Copyright 2020, Yahoo Inc.
  * Licensed under the Apache License, Version 2.0
  * See LICENSE file in project root for terms.
  */
 package com.yahoo.elide.security;
 
 import lombok.Getter;
+import java.security.Principal;
 
 /**
  * Wrapper for opaque user passed in every request.
  */
 public class User {
-    @Getter private final Object opaqueUser;
```

Comment:
Are there any known use cases that would not be supported by Principal?

Reply:
Not that I can think of.  Spring maps everything down to a subclass of Principal. JAXRS has both the SecurityContext and the Principal.  Where we have used JAXRS - we tend to put all the user bits into a subclass of Principal.